AnalyzerTools/src/MCCorrection.C

```diff
@@ -1070,6 +1076,74 @@ double MCCorrection::GetJetTaggingCutValue(JetTagging::Tagger tagger, JetTagging
 
 }
 
+void MCCorrection::SetupMCJetTagEff(){
+  cout<<"[MCCorrection::SetupMCJetTagEff] setting MCJetTagEff"<<endl;
+
+  TString datapath=getenv("DATA_DIR");
+  TString mcjetpath=datapath+"/"+DataEra+"/BTag/MeasureJetTaggingEfficiency_TTLL_TTLJ_hadded.root";
+  ifstream fcheck(mcjetpath);
+  if(!fcheck.good()){
+    cout<<"[MCCorrection::SetupMCJetTagEff] no "+mcjetpath<<endl;
+    return;
+  }
+  TFile fmcjet(mcjetpath);
+  // Denominator histogram setup first
+  vector<TString> jfs = {"B", "C", "Light"};
+  for(unsigned int i=0; i<jfs.size(); i++){
+    TString hden="Jet_"+DataEra+"_eff_"+jfs.at(i)+"_denom";
+    TH2F* this_hist=(TH2F*)fmcjet.Get(hden);
+    map_hist_mcjet[hden]=this_hist;
+    this_hist->SetDirectory(0);
+    cout<<"[MCCorrection::SetupMCJetTagEff] setting "<<hden<<endl;
+  }
+  // Numerator histogram setup and divided using "binomial option"
+  for(const auto& obj:*(fmcjet.GetListOfKeys())){
+    TH2F* this_hist=(TH2F*)((TKey*)obj)->ReadObj();
+    TString hnum=this_hist->GetName();
+    if(!hnum.Contains("num")) continue;
+    TString hden="";
+    if(hnum.Contains("_B_")) hden="Jet_"+DataEra+"_eff_B_denom";
+    else if(hnum.Contains("_C_")) hden="Jet_"+DataEra+"_eff_C_denom";
+    else hden="Jet_"+DataEra+"_eff_Light_denom";
+
+    this_hist->Divide(this_hist,map_hist_mcjet[hden],1.,1.,"b");
+    map_hist_mcjet[hnum]=this_hist;
+    this_hist->SetDirectory(0);
+    cout<<"[MCCorrection::SetupMCJetTagEff] setting "<<hnum<<endl;
+  }
+}
+
+double MCCorrection::GetMCJetTagEff(JetTagging::Tagger tagger, JetTagging::WP wp, int JetFlavor, double JetPt, double JetEta, int sys){
+
+  if(IsDATA) return 1.;
+
+  if(JetPt<20) JetPt = 20.;
+  if(JetPt>=1000.) JetPt = 999.;
+  if(JetEta>=2.5) JetEta = 2.49;
+  if(JetEta<-2.5) JetEta = -2.5;
+
+  TString jf = "";
+  if(JetFlavor == 5) jf = "B";
+  else if(JetFlavor == 4) jf = "C";
+  else if(JetFlavor == 0) jf = "Light";
+  else{
+    cout<<"[MCCorrection::GetMCJetTagEff] no JetFlavor"<<endl;
+    exit(EXIT_FAILURE);
+  }
+
+  double value = 1., error = 0., out = 1.;
+  TString hnum="Jet_"+DataEra+"_"+JetTagging::TaggerToString(tagger)+"_"+JetTagging::WPToString(wp)+"_eff_"+jf+"_num";
+  TH2F *this_hist = map_hist_mcjet[hnum];
+  int this_bin = this_hist->FindBin(JetPt,JetEta);
```

Comment:
This line should be fixed. Sorry All!!
int this_bin = this_hist->FindBin(JetEta,JetPt);
Since this mistake, MC eff was always 0.0001